src/librustc_typeck/check/method/suggest.rs

```diff
@@ -424,10 +424,12 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                 }
 
                 if !unsatisfied_predicates.is_empty() {
-                    let bound_list = unsatisfied_predicates.iter()
+                    let mut bound_list = unsatisfied_predicates.iter()
                         .map(|p| format!("`{} : {}`", p.self_ty(), p))
-                        .collect::<Vec<_>>()
-                        .join("\n");
+                        .collect::<Vec<_>>();
+                    bound_list.sort();
+                    bound_list.dedup_by(|a, b| a == b);  // #35677
```

Comment:
isn't this equivalent to 

```suggestion
                    bound_list.dedup();  // #35677
```

Reply:
It is, didn't see it when searching for it in the docs, it turns out it was just out of order :)